donkeycar/management/tub_gui.py

```diff
@@ -0,0 +1,598 @@
+#!/usr/bin/env python3
```

Comment:
General comment: Reviewing UI code is quite hard. 

Especially given I see a few `row`s of UI being built with the implementation that looks similar enough to refactor. Also the calls to `decompose` some information stored in strings, I am not sure why its being done that way. 

I am okay with leaving it the way it is - but it might be pretty hard to maintain this going forward. OOC : Any reason why you did not choose to extend the web UI ?

Reply:
The web UI was very hard to change. In my fork I had added some sub-functionality, like displaying more data  (`user/angle, user/throttle, car/speed, car/distance`, etc), in the html, but it was a trial and error exercise. I gave up on the idea to hook that up dynamically to the tub content - for example `car/speed, car/distance` is not something other people might have in their tubs. Hence the new UI dynamically reads the fields from the tub. All this functionality is easy in the new tub UI. Additionally handling large tubs is super performant. You will need a lot of patience if you want to edit a 300k tub in the web UI.  I spent too many hours in front of the web UI :-/

I agree that the code in the web UI can be better documented and I can do some small re-factorings and break it into smaller parts. For now, I thought we just offer this as an alternative. The `decompose` function is used for data stored in vectors, which we support in the tub. E.g.  my tubs store IMU data in two 3-d vectors instead of 6 scalars. The UI will support both and decompose the vectors and you can individually view only some components. 

I am also thinking of porting the UI to kivy, but this will only affect the look and not the functionality. Kivy is quite sleek, also on Linux, and uses graphics card OpenGL rendering.